x/slashing/handler_sidechain.go

```diff
@@ -16,17 +19,35 @@ func handleMsgBscSubmitEvidence(ctx sdk.Context, msg MsgBscSubmitEvidence, k Kee
 		return ErrInvalidSideChainId(DefaultCodespace).Result()
 	}
 
+	c := SideChainIdFromText(sideChainId)
+	chainID := big.NewInt(c)
+	// Comment out the 2 lines above and use this for testing (local chainID)
+	// chainID := big.NewInt(48)
+
 	header := ctx.BlockHeader()
-	sideConsAddr, err := msg.Headers[0].ExtractSignerFromHeader()
+
+	var sideConsAddr bsc.Address
+	var sideConsAddr2 bsc.Address
+
+	if sdk.IsUpgrade(sdk.BEP174) {
+		sideConsAddr, err = msg.Headers[0].ExtractSignerFromHeader(chainID)
```

Comment:
Why don't we extract the consaddr2 here as well

Reply:
I was following the format of the original code but I can change it